src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_enkf_update.F90

```diff
@@ -2932,6 +2935,114 @@ end subroutine write_smapL4SMaup
+    !                         - surface_temp_analysis
+    !                         - soil_temp_layer1_analysis
+
+    implicit none
+    
+    character(*),          intent(in) :: option
+    integer,                intent(in) ::  N_ens, N_catl 
+    type(cat_param_type),   dimension(N_catl),        intent(in)    :: cat_param
+    type(cat_progn_type),   dimension(N_catl,N_ens),  intent(in)    :: cat_progn
+    type(cat_progn_type),   dimension(N_catl),        intent(inout)    :: diagaup
+
+    real, dimension(      N_catl, 5)     :: tile_mean_l   !out diagaup(:, 5)=tile_mean_l(:,5) 
+    real, dimension(      N_catl, 5)     :: tile_std_l
+
+    real, dimension(      N_catl)        :: srfexc, rzexc, catdef
+    real, dimension(      N_catl)        :: ar1,    ar2,   ar4
+    real, dimension(N_gt, N_catl)        :: tp
+    real, dimension(      N_catl, N_ens) :: sfmc,   rzmc,  prmc, tsurf, tp1
+    integer   :: n_e, j, n 
+    character(len=*), parameter :: Iam = 'get_diagaup'
+    character(len=400) :: err_msg
+
+!----------------------
+
+        do n_e=1,N_ens
+
+          srfexc = cat_progn(:,n_e)%srfexc 
+          rzexc  = cat_progn(:,n_e)%rzexc 
+          catdef = cat_progn(:,n_e)%catdef  
+
+          call catch_calc_soil_moist(                                             &
+               N_catl,                                                            &
+               cat_param%vegcls, cat_param%dzsf,   cat_param%vgwmax,              &
+               cat_param%cdcr1,  cat_param%cdcr2,  cat_param%psis,                &
+               cat_param%bee,    cat_param%poros,  cat_param%wpwet,               &
+               cat_param%ars1,   cat_param%ars2,   cat_param%ars3,                &
+               cat_param%ara1,   cat_param%ara2,   cat_param%ara3,                &
+               cat_param%ara4,   cat_param%arw1,   cat_param%arw2,                &
+               cat_param%arw3,   cat_param%arw4,                                  &
+               srfexc,           rzexc,            catdef,                        &
+               ar1,  ar2,  ar4, sfmc(:,n_e), rzmc(:,n_e), prmc(:,n_e) ) 
+
+          call catch_calc_tsurf( N_catl,                                          &
+               cat_progn(:,n_e)%tc1, cat_progn(:,n_e)%tc2, cat_progn(:,n_e)%tc4,  &
+               catprogn2wesn(N_catl,cat_progn(:,n_e)),                            &
+               catprogn2htsn(N_catl,cat_progn(:,n_e)),                            &
+               ar1,  ar2,  ar4,                                                   &
+               tsurf(:,n_e) )
+
+          call catch_calc_tp( N_catl, cat_param%poros,                            &
+               catprogn2ghtcnt(N_catl,cat_progn(:,n_e)), tp )
+
+          tp1(:,n_e) = tp(1,:) + MAPL_TICE
+
+       end do  !n_e N_ens 
+
+       ! compute ensemble mean values and std-dev
+       call row_std( N_catl, N_ens, sfmc,  tile_std_l(:,1), tile_mean_l(:,1) )
```

Comment:
The LDASsa "aup" files include not only the (ensemble-mean) FCST and ANA variables but also the ensemble std-dev.  The above edits ignore the possibility of adding ens-std-dev values into the "lndfcstana" file, which we will probably want later on...
For now perhaps just keep it to the ens mean, but keep in mind that later on we'll probably want to add the ens-std-dev as well.  We would then have to decide again whether to do the ens stats here or in the Ens Grid Comp.